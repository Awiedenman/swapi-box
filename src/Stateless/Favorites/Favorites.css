.favorites-display {
  position: absolute;
  top: 15px;
  right: 10px;
  height: 22px;
  width: 116px;
  border: 2px solid #ffff00;
  padding: 10px;
  background: #ffffff
}

.fav-counter {
  border: 1px solid #000000;
  padding: 5px 10px 5px 5px;
  margin: 0 0 0 15px;
}
button {
  border: none;
}

button:hover {
  color: red;
  border-bottom: red;
}