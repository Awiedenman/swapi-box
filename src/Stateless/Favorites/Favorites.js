import React from 'react';
import './Favorites.css';
import PropTypes from 'prop-types';

const Favorites = ( { count, showFavorites } ) => {
  let name = 'favorites';
  return(
    <div className="favorites-display">
      <button className="showFavorites-button" onClick={ () => showFavorites( name )}> Favoites 
        <span className="fav-counter"> { count } </span>
      </button>
    </div>
  )
}

export default Favorites;