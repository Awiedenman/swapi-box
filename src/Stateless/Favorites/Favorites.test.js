import React from 'react';
import Favorites from './Favorites';
import { shallow } from 'enzyme';

describe('Favorites', () => {
  const mockCount= 1;
  const mockShowFavorites = jest.fn();
  const wrapper = shallow(
    < Favorites
      count = { mockCount }
      showFavorites = {mockShowFavorites }
    />
  );
  it('should match the snapshot of the favorites component ont he page', () => {
    expect( wrapper ).toMatchSnapshot();

  });
});