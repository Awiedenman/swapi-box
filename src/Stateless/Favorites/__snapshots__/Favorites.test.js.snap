// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match the snapshot of the favorites component ont he page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    count={1}
    showFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="showFavorites-button"
        onClick={[Function]}
>
         Favoites
        <span
                className="fav-counter"
        >
                 
                1
                 
        </span>
</button>,
      "className": "favorites-display",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " Favoites",
          <span
            className="fav-counter"
>
             
            1
             
</span>,
        ],
        "className": "showFavorites-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        " Favoites",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              1,
              " ",
            ],
            "className": "fav-counter",
          },
          "ref": null,
          "rendered": Array [
            " ",
            1,
            " ",
          ],
          "type": "span",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="showFavorites-button"
          onClick={[Function]}
>
           Favoites
          <span
                    className="fav-counter"
          >
                     
                    1
                     
          </span>
</button>,
        "className": "favorites-display",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Favoites",
            <span
              className="fav-counter"
>
               
              1
               
</span>,
          ],
          "className": "showFavorites-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          " Favoites",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                1,
                " ",
              ],
              "className": "fav-counter",
            },
            "ref": null,
            "rendered": Array [
              " ",
              1,
              " ",
            ],
            "type": "span",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
